src/types.ts

```diff
@@ -6,6 +67,7 @@
  */
 export type RemainingPropsAndTestOverrides<
   ComponentType extends React.ComponentType,
-  BaseProps
-> = Omit<React.ComponentProps<ComponentType>, keyof BaseProps> &
-  Partial<React.ComponentProps<ComponentType>>;
+  BaseProps extends Partial<FullProps<ComponentType>>
+> =
+  | Omit<FullProps<ComponentType>, keyof BaseProps>
+  | Pick<FullProps<ComponentType>, keyof Partial<FullProps<ComponentType>>>;
```

Comment:
Heh, Josh's initial assumption proved right. Coming to this file I'm going in circles piecing this logic together. I know we commented the hell out of it. I'm wondering if at even a higher level, we can either:

Put a summary in this pr of how all the interacting pieces come together in the pr description

Add documentation to notion on how we currently use typescript in RTL, and include that same summary 

The good thing is that this is forcing me to look up a lot of TS utility functions etc and also understand the multiple levels of extending other types. Maybe what I'm looking for is a meta summary that'll allow me to tease apart these comments better on my own!

Reply:
Some of the gaps for me, in summary are:

- Insufficient practice/understanding of some utility types to make sense of this off the bat
- Insufficient exposure to specific operators, to understand helper functions fully (i.e. `keyof`)